diff --git a/src/duc/cmd-cgi.c b/src/duc/cmd-cgi.c
index ae2fbe8..ded0bc5 100644
--- a/src/duc/cmd-cgi.c
+++ b/src/duc/cmd-cgi.c
@@ -214,7 +214,8 @@ static void print_script(const char *path)
 		"        var rect = img.getBoundingClientRect(img);\n"
 		"        var x = e.clientX - rect.left;\n"
 		"        var y = e.clientY - rect.top;\n"
-		"        window.location = '?x=' + x + '&y=' + y + '&path=");
+		"        var db = (new URL(location.href)).searchParams.get('db')\n"
+		"        window.location = '?db=' + db + '&x=' + x + '&y=' + y + '&path=");
 	print_html(path);
 	printf( "';\n"
 		"      }\n"
@@ -320,8 +321,8 @@ static void do_index(duc *duc, duc_graph *graph, duc_dir *dir)
 			dir = dir2;
 			path = duc_dir_get_path(dir);
 			printf("Status: 302 Found\n");
-			printf("Location: ?path=%s\n", path);
-			printf("URI: ?path=%s\n", path);
+			printf("Location: ?db=%s&path=%s\n", opt_database, path);
+			printf("URI: ?db=%s&path=%s\n", opt_database, path);
 			printf("Connection: close\n");
 			printf("Content-type: text/html\n\n");
 			printf("\n");
@@ -361,7 +362,7 @@ static void do_index(duc *duc, duc_graph *graph, duc_dir *dir)
 		duc_human_size(&report->size, st, 0, siz, sizeof siz);
 
 		printf("  <tr>\n");
-		printf("   <td><a href=\"%s&path=", url);
+		printf("   <td><a href=\"%s&db=%s&path=", url, opt_database);
 		print_cgi(report->path);
 		printf("\">");
 		print_html(report->path);
@@ -406,7 +407,7 @@ static void do_index(duc *duc, duc_graph *graph, duc_dir *dir)
 			printf("  <tr><td class=name>");
 
 			if(e->type == DUC_FILE_TYPE_DIR) {
-				printf("<a href=\"%s&path=", url);
+				printf("<a href=\"%s&db=%s&path=", url, opt_database);
 				print_cgi(path);
 				printf("/");
 				print_cgi(e->name);
