diff --git a/src/duc/cmd-cgi.c b/src/duc/cmd-cgi.c
index ded0bc5..a613333 100644
--- a/src/duc/cmd-cgi.c
+++ b/src/duc/cmd-cgi.c
@@ -33,6 +33,8 @@ static bool opt_gradient = false;
 static char *opt_footer = NULL;
 static double opt_fuzz = 0.7;
 static char *opt_header = NULL;
+static char *opt_error_db = NULL;
+static char *opt_error_path = NULL;
 static int opt_levels = 4;
 static char *opt_palette = NULL;
 static bool opt_tooltip = false;
@@ -277,6 +279,7 @@ static void print_html_header(const char *path)
 		 "\n"
 		 "<!DOCTYPE html>\n"
 		 "<head>\n"
+		 "  <title>Storage Analyzer</title>\n"
 		 "  <meta charset=\"utf-8\" />\n"
 		 );
   
@@ -500,8 +503,12 @@ static int cgi_main(duc *duc, int argc, char **argv)
 
         r = duc_open(duc, opt_database, DUC_OPEN_RO);
         if(r != DUC_OK) {
-		printf("Content-Type: text/plain\n\n");
-                printf("%s\n", duc_strerror(duc));
+		printf("Status: 302 Found\n");
+		printf("Location: %s?db=%s\n", opt_error_db, opt_database);
+		printf("URI: %s?db=%s\n", opt_error_db, opt_database);
+		printf("Connection: close\n");
+		printf("Content-type: text/html\n\n");
+		printf("\n");
 		return -1;
         }
 
@@ -510,9 +517,12 @@ static int cgi_main(duc *duc, int argc, char **argv)
 	if(path) {
 		dir = duc_dir_open(duc, path);
 		if(dir == NULL) {
-			printf("Content-Type: text/plain\n\n");
-			printf("%s\n", duc_strerror(duc));
-			print_html(path);
+			printf("Status: 302 Found\n");
+			printf("Location: %s?db=%s&path=%s\n", opt_error_path, opt_database, path);
+			printf("URI: %s?db=%s&path=%s\n", opt_error_path, opt_database, path);
+			printf("Connection: close\n");
+			printf("Content-type: text/html\n\n");
+			printf("\n");
 			return -1;
 		}
 	}
@@ -562,6 +572,9 @@ static struct ducrc_option options[] = {
 	{ &opt_fuzz,      "fuzz",       0,  DUCRC_TYPE_DOUBLE, "use radius fuzz factor when drawing graph [0.7]" },
 	{ &opt_gradient,  "gradient",   0,  DUCRC_TYPE_BOOL,   "draw graph with color gradient" },
 	{ &opt_header,    "header",     0,  DUCRC_TYPE_STRING, "select HTML file to include as header" },
+	{ &opt_error_db,  "db-error",   0,  DUCRC_TYPE_STRING, "select HTML file to show in case of a DB parameter error" },
+	{ &opt_error_path,"path-error", 0,  DUCRC_TYPE_STRING, "select HTML file to show in case of a path parameter error" },
+	{ &opt_header,    "header",     0,  DUCRC_TYPE_STRING, "select HTML file to include as header" },
 	{ &opt_levels,    "levels",    'l', DUCRC_TYPE_INT,    "draw up to ARG levels deep [4]" },
 	{ &opt_list,      "list",       0,  DUCRC_TYPE_BOOL,   "generate table with file list" },
 	{ &opt_palette,   "palette",    0,  DUCRC_TYPE_STRING, "select palette",
